core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -61,6 +61,9 @@ object Automation {
             for (unique in localUniqueCache.forCityGetMatchingUniques(city, UniqueType.UnhappinessFromPopulationTypePercentageChange))
                 if (unique.params[1] == "Specialists" && city.matchesFilter(unique.params[2]))
                     yieldStats.happiness -= (unique.params[0].toFloat() / 100f)  // relative val is negative, make positive
+            yieldStats.science *= 1.5f // we want to be working scientists
```

Comment:
This is more of a player preference, can we leave this out?

Reply:
The base value of science is pretty low. We could increase it to make the AI work scientist slots, but this would lead to them getting stuck working trade posts and Krakatoa. The 'cleanest' way to properly implement this, is to make the AI look up the abilities of the great person corresponding to the great person points and value the slots accordingly, but is a bit too complex for me to implement. It seems multiplying the value of science by 1.5 is approximately good enough to let the AI rank scientists as the highest of the specialists.

Reply:
Great scientists in the base game are generally regarded as the best great persons. In Rekmod, each type of great person is in their own gpp pool, while manufactories and customs houses are much better, You could argue engineer or merchant slots are the better slots, but when I visit the Lekmod discord, people there are obsessed with discussing scientists management, so I think it holds true across base rulesets.

Reply:
> This is more of a player preference, can we leave this out?
What's your proposal on letting the AI work the right specialist slots?

Reply:
The prerequisite questions are:

1. How do we figure out the best research slot? 
2. Do we want all the AI Civs to choose one specialist type?
3. Do we want to choose a specific strategy that we deem best and try to get the AI and/or players to use it?

For the first question, if we want to dynamically determine this we'd need some sort of unique evaluation function that can rank the uniques together given the current game state and give us a value. It seems like we are having quite some trouble with this. Choose the best one for G&K or some other version would not play well with mods which could re-balance or completely change them.

For the second and third questions, each nation has their own personality (implemented in  #10939), meaning we encourage them to be different from each-other. So why would we try and override that? Making decisions for players doesn't tend to go well, this PR for example was the result of a lighter version of that.

Reply:
> For the first question, if we want to dynamically determine this we'd need some sort of unique evaluation function that can rank the uniques together given the current game state and give us a value. It seems like we are having quite some trouble with this. Choose the best one for G&K or some other version would not play well with mods which could re-balance or completely change them.

True, and also, uniques also tend to get renamed quite a bit. The simplest (but not fool-proof) way to do this would be to adjust the evaluation of science, as the value of GS points (which are good in both bas game and Rekmod at least) is implicit in it.

> For the second and third questions, each nation has their own personality (implemented in #10939), meaning we encourage them to be different from each-other. So why would we try and override that?

If I play test with the current implementation, a custom Default does show to rank scientists as the highest of the specialists, but doesn't work all of them all the time. Songhai keeps prioritizing merchants, and India keeps prioritizing artists (I haven't tested for every civ).

> Making decisions for players doesn't tend to go well, this PR for example was the result of a lighter version of that.

The default focus is currently not very good, so I've got to manually assign all my citizens. From playtesting, it appears this PR solves some of it, but I'll likely give it some more fine-tuning in the future.